Add mobile notifications and adapt its design

Tbaut opened the pull request:
closes #1881
Did my best to stick to it the mock. there might be slight differences like shadows on the notification button (the mock has some, not mine), or on the notification panel (the mock doesn't have any, mine has some). But those are deliberate to keep consistency for the buttons, and to show that the notification panel is above the rest, otherwise it wasn't clear IMHO. Also I did not expand the notification panel all the way so that it's easier to click out. lmk your thoughts.

ping @serenaho if you want to play with it as well and tell me what I forgot/missed.

How to get a notification:
- go to settings > subscriptions
- change plan and chose a more expensive one
- choose annual billing and crypto billing
- go until the screen where you can chose the currency
- from that point in time you should have a "outstanding invoice" notification, for 1 hour.
---
Submission checklist: 

<!-- Remove anything below that is not applicable -->   

#### Layout
- [x] Change looks good in the desktop web ui
- [x] Change looks good in the mobile web ui

#### Theme
- [x] Components / elements inspected in light mode 
- [x] Components / elements inspected in dark mode

Tbaut committed:
search bar on mobile leaving the hamburger visible

Tbaut committed:
CSS for mobile

Tbaut committed:
lots of fiddling

Tbaut committed:
dark mode fix

Tbaut commented on packages/files-ui/src/Components/Modules/SearchModule.tsx:
```diff
@@ -1,19 +1,8 @@
-  useHistory,
-  useToasts
-} from "@chainsafe/common-components"
+  // ArrowLeftIcon, Button,
```
```suggestion
```

Tbaut committed:
Update packages/files-ui/src/Components/Modules/SearchModule.tsx

actions-user committed:
lingui extract

Tbaut committed:
Merge branch 'dev' into mnt/tbaut-notif-mobile-1881

serenaho commented:
I reviewed the changes and things are looking good! Mostly small style fixes here and there (which unfortunately I cannot go back to review because of the 60 min hold) but these are the elements I noticed first:

1. The padding of the Notification button is currently bigger compared to the other buttons **and** the red circle with the "1" doesn't go away when you click on it.

<img width="191" alt="Screen Shot 2022-01-28 at 12 01 14 PM" src="https://user-images.githubusercontent.com/91699405/151613601-fb53d34c-791d-40f0-9baf-100562259d17.png">


2. This modal has two "Go back" options (remove the underlined one)
![Screen Shot 2022-01-28 at 11 28 15 AM](https://user-images.githubusercontent.com/91699405/151613728-cd248412-80de-48f6-984c-c4eb9e904bb9.png)

3. The hover for Notifications looks a bit slower compared to the other buttons

https://user-images.githubusercontent.com/91699405/151614038-ab2261ae-c20a-497e-8f65-1f3fa2a2bdbb.mov

serenaho commented:
I also realized that there aren't currently any options to cancel a plan change once you start the crypto payment process. 

I know the 60 minute timer eventually times out, but there's currently no way for the user to know that their plan status reverts to normal if they just wait (they might try to contact us).

Is it possible to cancel it once we come back to Settings like below?

![image](https://user-images.githubusercontent.com/91699405/151615002-52c2b319-d045-418d-9dd6-5487eee41b05.png)

If not, it may be worth it to put a message somewhere saying that the "Awaiting payment" status will go away.

_*Side note, this might sound like an edge case but I imagine some users might be curious to see what the crypto payment process looks like and not realize it triggers something_

asnaith commented:
I noticed something minor - when tapping to invoke the notification panel the icon turns blue but if you tap on the icon to dismiss the panel the icon remains blue until you tap the screen again elsewhere. 

I think the colour should update on the tap that hides the panel rather than on the proceeding screen tap after that action. 

https://user-images.githubusercontent.com/3982665/151636874-419a48d2-d2ea-489c-8cf6-5a5d11e180a6.mp4

I also noticed the persisting notification count number but I'm guessing that's because the actual notification persists too, I'm guessing this behavior is expected until / if we have a clear-all button on the panel?

Tbaut committed:
Merge branch 'dev' of github.com:ChainSafe/ui-monorepo into mnt/tbaut-notif-mobile-1881

Tbaut committed:
Merge branch 'mnt/tbaut-notif-mobile-1881' of github.com:ChainSafe/ui-monorepo into mnt/tbaut-notif-mobile-1881

Tbaut committed:
padding

Tbaut committed:
remove button for ntofications

Tbaut committed:
background

Tbaut commented:
Thanks for the review, there's a lot to unpack here, I'll start with what I addressed:
@asnaith 
> when tapping to invoke the notification panel the icon turns blue but if you tap on the icon to dismiss the panel the icon remains blue until you tap the screen again elsewhere.

We were using a button so far, and this was somehow the standard behavior for buttons, linked to our "focus" styling.
I've set this notification to not be a button anymore, and made the style match with the others buttons, that's quite a lot of changes, for a small visual differences, so we should take a closer look.

@serenaho 
> The hover for Notifications looks a bit slower compared to the other buttons

I've a hard time reproducing, but I think I fixed it by removing the transition, can you check?

> The padding of the Notification button is currently bigger compared to the other buttons

That should be fixed now

> and the red circle with the "1" doesn't go away when you click on it.

This is not something I've set here, since I only took a particular care about mobile notifications. As Andrew mentioned, this is indeed something that we want, since this notification is very important. Unless you pay or let the 60min go by, you won't be able to dismiss it!

> which unfortunately I cannot go back to review because of the 60 min hold

You can login with another user to have no notification again

> This modal has two "Go back" options (remove the underlined one)

So the second "go back" is to close the "add card" section. We should find a better wording, but they are different.

> I also realized that there aren't currently any options to cancel a plan change once you start the crypto payment process...

This is also a conversation that should be had outside of this PR, since it's not related :smile:

Tbaut commented on packages/files-ui/src/locales/fr/messages.po:
```diff
@@ -596,12 +596,18 @@ msgstr "Non merci"
 msgstr "Aucun utilisateur n'a été trouvé pour cette requête."
 
+msgid "Notifications"
+msgstr ""
```
```suggestion
msgstr "Notifications"
```

Tbaut committed:
Update packages/files-ui/src/locales/fr/messages.po

tanmoyAtb reviewed (changes requested):
Code looks spotless 💯 

I would suggest a few style changes regarding the notification dropdown itself. 

<img width="358" alt="Screenshot 2022-01-31 at 10 23 06 PM" src="https://user-images.githubusercontent.com/10713353/151832204-f06165a6-7a37-46ed-9184-b4cd2431a050.png">
<img width="373" alt="Screenshot 2022-01-31 at 10 27 20 PM" src="https://user-images.githubusercontent.com/10713353/151832753-35ea3e57-c885-443b-8106-cbec2854c0c5.png">

1. remove the left side padding, the notification text should align with `This week` heading.
2. Allow enough space for the space for the "time" text
3. Time text should also be right aligned. 

One functionality update on my mind. I expect the notification dropdown to close when I click on a notification for example to go to the billing page. 
We can add a `closeOnClick: boolean` key to the notification object.

FSM1 reviewed (approved):
great stuff @Tbaut  🎉

asnaith commented:
> We were using a button so far, and this was somehow the standard behavior for buttons, linked to our "focus" styling. I've set this notification to not be a button anymore, and made the style match with the others buttons, that's quite a lot of changes, for a small visual differences, so we should take a closer look.
> 

I'm still seeing a difference in behavior between this notification icon and the buttons to the left of it. If you see the way to "start a team" button works - it stays blue whilst the dialog is in view but the blue background disappears when the modal is dismissed, this is what I had in mind for the notification icon and panel (sorry, I know this sounds nitpicky but thought consistency is best).

https://user-images.githubusercontent.com/3982665/151850450-c64ee13b-9467-4eed-a7b3-b974639b3012.mov

Tbaut committed:
Merge branch 'dev' into mnt/tbaut-notif-mobile-1881

Tbaut commented:
> I'm still seeing a difference in behavior between this notification icon and the buttons

Sure, because now it's not a button any more. A button remains in focus when you click it, until you click away. This is exactly what you do in the video above. And this was also the original issue you mentioned. if you click the button while it's in focus, it acts as a "click outside" for the dropdown, and closes it, but since you just clicked the button, it's still focused and blue.

All in all, I'm not sure what's better, or if I should revert. I'd be willing to forget about the confusion it creates, as small as it can be.

Tbaut committed:
larger notifs, text alignment and autoclose

Tbaut commented:
agreed with your comments @tanmoyAtb I changed the padding to stick to the design (so I added a padding to the time header), aligned the time right, and leveraged the `autoclose` from the dropdown to make sure the panel closes when we click on something, even if that's a menu element, a click on anything closes the notification... It's a quick yet not dirty fix, hope it doesn't get in the way in the future. Now this PR has been open for a while, and we keep adding things, I'd like to go on :)

tanmoyAtb commented on packages/common-components/src/MenuDropdown/MenuDropdown.tsx:
```diff
@@ -258,6 +255,8 @@ const MenuDropdown = ({
     onScroll: managePosition
   })
 
+  console.log("menuItems", menuItems)
```
```suggestion
```

Tbaut replied:
oops thanks

tanmoyAtb reviewed (approved):
Looks good to go to me

Tbaut committed:
Update packages/common-components/src/MenuDropdown/MenuDropdown.tsx

Co-authored-by: Tanmoy Basak Anjan <tanmoy3399@gmail.com>

Tbaut committed:
Merge branch 'dev' into mnt/tbaut-notif-mobile-1881

asnaith commented:
> All in all, I'm not sure what's better, the current (non button) or if I should revert. I'd be willing to forget about the confusion it creates, as small as it can be. In any case what we see is the normal behavior of a button.. and a confusion created by the fact that we have a focus looking the same as a hover. I believe we did this for the login originally, but that's another story.

@Tbaut 👍 Ok, sorry to cause any confusion over something so minor. Happy to go with whatever you deem best either the current way or revert back to a button.

Tbaut committed:
Merge branch 'dev' into mnt/tbaut-notif-mobile-1881

Tbaut committed:
is that great but convoluted?

tanmoyAtb commented:
> I achieved what is IMHO good UX, at the cost of a somewhat more complex dropdown component though. I'd need an ack from @tanmoyAtb or @FSM1 before merging just for sanity and making sure we agree on the approach.
> 
>  active-notif.mp4

The hack looks sane to me. Keeping a state to know the dropdown is open or now looks like the straight forward approach. 👍
